onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -297,35 +298,49 @@ def enketo(self, request, *args, **kwargs):
 
     def destroy(self, request, *args, **kwargs):
         instance_ids = request.data.get('instance_ids')
+        delete_all_submissions = request.data.get('delete_all', False)
         self.object = self.get_object()
 
         if isinstance(self.object, XForm):
-            if not instance_ids:
+            if not instance_ids and delete_all_submissions == 'False':
```

Comment:
Could we convert the `delete_all_submissions` to a bool using the `strtobool` function. An example of this is [here](https://github.com/onaio/onadata/blob/ded0f36bf17db95af42ee1454487e9a769edfa57/onadata/apps/api/viewsets/project_viewset.py#L138-L139)

Reply:
Updated that. thanks